addons/OPT/GPS/fn_addUnitToGPS.sqf

```diff
@@ -22,99 +20,100 @@
 * No
 * 
 * Example:
-* _iconID = [player, "Icon_Player"] call FUNC(addUnitToGPS);
+* [player, "Icon_Player"] call FUNC(addUnitToGPS);
 */
 
 #include "macros.hpp"
 
 
-params ["_newUnit", "_iconId"];
 
-// Decide on the right color
-private _color = if (CLib_Player isEqualTo _newUnit) then {
-    COLOR_PLAYER_UNIT
-    } else {
-        if (group CLib_Player isEqualTo group _newUnit) then {
-            COLOR_OWN_GROUP
-        } else {
-            COLOR_SIDE
-        }
+[{
+    params ["_newUnit"];
+    DUMP("UNIT ICON ADDED");
+    DUMP(_newUnit);
+
+    if (!([_newUnit] call FUNC(isUnitVisible))) exitWith {
+        DUMP(format ["Unit %1 is not visible", _newUnit]);
     };
+    private _iconID = _newUnit call FUNC(getUnitIconID);
+    // Decide on the right color
+    private _color = if (CLib_Player isEqualTo _newUnit) then {
+        COLOR_PLAYER_UNIT
+        } else {
+            if (group CLib_Player isEqualTo group _newUnit) then {
+                COLOR_OWN_GROUP
+            } else {
+                COLOR_SIDE
+            }
+        };
 
+    private _specialTexture = getText (configFile >> "CfgVehicles" >> typeOf _newUnit >> "icon");
+    private _texture = format ["\A3\ui_f\data\map\vehicleicons\%1_ca.paa", if (_specialTexture == "") then {"iconMan"} else {_specialTexture}];
+    private _text = [_newUnit] call CFUNC(name);
 
-// Icon is pulled from opt_characters client mod.
-private _specialicon = getText (configFile >> "CfgVehicles" >> typeOf _newUnit >> "icon");
-private _icon = format ["\A3\ui_f\data\map\vehicleicons\%1_ca.paa", if (_specialicon == "") then {"iconMan"} else {_specialicon}];
-
-// Default format
-private _width = 20;
-private _height = 20;
-private _angle = _newUnit;
-
-
-// Define the icon shown on the map.
-private _unitIcon = ["ICON", 
-    _icon, // 1: Texture <String>
-    _color, // 2: Color <Array> [r,g,b,a]
-    _newUnit, // 3: Position <MapGraphicsPosition> // We place the object itself here. This allows us hacky access later to retrieve variables of it for conditional onEachFrame styling.
-    _width, // 4: Width <Number>
-    _height, // 5: Height <Number>
-    _newUnit, // 6: Angle <Number>
-    "", // 7: Text <String>
-    1, // 8: Shadow <Boolean/Number>
-    0.08, // 9: Text Size <Number>
-    "RobotoCondensed", // 10: Font <String>
-    "right", // 11: Align <String>
-    {
-        // Only show if medic?
-        //if ((typeof CLib_Player == "OPT_NATO_Sanitaeter" || typeof CLib_Player == "OPT_CSAT_Sanitaeter") && {_newUnit getVariable ["FAR_isUnconscious", 0] == 1}) then {
-        //Show wounded marker if downed
-        if (_position getVariable ["FAR_isUnconscious", 0] == 1) then {
+    if (_newUnit getVariable ["FAR_isUnconscious", 0] == 1) then {
+        _texture = "\A3\ui_f\data\igui\cfg\revive\overlayicons\u100_ca.paa";
+        if (_newUnit getVariable ["FAR_IsStabilized", 0] == 1) then {
+            _text = format ["%1 (%2)", [_newUnit] call CFUNC(name), "Stabilisiert" ];
```

Comment:
`[_newUnit] call CFUNC(name)` ist hier unnötig, da das Ergebnis dieses Aufrufs zu diesem zeitpunkt bereits in `_text` steht. Würde ich aber auf jeden Fall noch als Kommentar mit dazu schreiben...
```
_text = format ["%1 (%2)", _text, "Stabilisiert" ];
```